app/lib/actions/candlepin/consumer/bind_entitlement.rb

```diff
@@ -0,0 +1,30 @@
+module Actions
+  module Candlepin
+    module Consumer
+      class BindEntitlement < Candlepin::Abstract
+        middleware.use Actions::Middleware::KeepCurrentUser
+
+        input_format do
+          param :organization_id, Integer
+          param :pools, Array
+        end
+
+        def plan(organization_id = nil, pools = [])
+          sequence do
+            plan_self(organization_id: organization_id, pools: pools)
```

Comment:
Will this blow up if I give nonsense pool ids?

Reply:
Candlepin handles the nonsense. It does a lookup on the pool ID; if it doesn't exist or isn't valid, we get a NotFound error back.

Reply:
It seemed like a good idea to me to rescue from that in my Delete action - do you think we shouldn't ? The reason I think it's good is to avoid resource locking problems on subsequent actions issued through the API

Reply:
Yeah, that sounds like a good plan to me.

Reply:
Actually, there is a mechanism already in place that handles these candlepin errors. We can just let them be raised, and all is reported correctly. https://github.com/Katello/katello/blob/6aca54157e579d6312a9e4a49609df0ffa1685b1/app/lib/actions/middleware/propagate_candlepin_errors.rb

It comes out looking like this: `Katello::Errors::CandlepinError: Pool with id 98c60c4fc0e0160c50f7eea0411 could not be found.`